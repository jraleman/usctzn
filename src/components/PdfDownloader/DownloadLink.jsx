import React from 'react';
import styled from 'styled-components';

const DownloadLink = ({ src = '', label = '' }) => {
    return (
        <Link href={src} download>
            ⏬ {label} ⏬
        </Link>
    );
};

const Link = styled.a`
    display: block;
    text-decoration: none;
    text-align: center;
    color: red;
    font-family: 'Helvetica', sans-serif;
    padding: 12px;
`;

export default DownloadLink;
