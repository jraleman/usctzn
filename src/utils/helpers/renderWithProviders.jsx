import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { lightTheme } from '../styles/theme';

const renderWithProviders = (Component) => {
    render(
        <ThemeProvider theme={lightTheme}>
            {Component}
        </ThemeProvider>
    )
};

export default renderWithProviders;
