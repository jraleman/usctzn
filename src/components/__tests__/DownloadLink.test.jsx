import React from 'react';
import renderWithProviders from '../../utils/helpers/renderWithProviders';
import DownloadLink from '../DownloadLink';

describe('DownloadLink', () => {
    it('renders', () => {
        const component = renderWithProviders(<DownloadLink />);
        expect(component).not.toBeNull();
        expect(component).toMatchSnapshot();
    });
});
