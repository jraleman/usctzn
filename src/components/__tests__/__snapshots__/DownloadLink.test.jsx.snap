// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadLink renders 1`] = `undefined`;
